tests/unit/app/endpoints/test_ols.py

```diff
@@ -139,17 +139,32 @@ def test_validate_question_on_validation_error(validate_question_mock, load_conf
         ols.validate_question(conversation_id, llm_request)
 
 
+@patch("ols.utils.config.question_redactor")
+def test_question_filter_on_redact_error(mock_redact_question, load_config):
+    """Test conversation request API endpoint."""
+    conversation_id = suid.get_suid()
+    query = "Tell me about Kubernetes"
+    llm_request = LLMRequest(query=query, conversation_id=conversation_id)
+    mock_redact_question.redact_question.side_effect = Exception
+    with pytest.raises(HTTPException, match="Error while redacting query"):
+        ols.redact_question(conversation_id, llm_request)
+
+
 # TODO: distribute individual test cases to separate test functions
 @patch("ols.src.query_helpers.question_validator.QuestionValidator.validate_question")
 @patch("ols.src.query_helpers.docs_summarizer.DocsSummarizer.summarize")
 @patch("ols.utils.config.conversation_cache.get")
+@patch("ols.utils.config.QuestionFilter.redact_question")
 def test_conversation_request(
+    mock_question_redactor,
     mock_conversation_cache_get,
     mock_summarize,
     mock_validate_question,
-    load_config,
 ):
     """Test conversation request API endpoint."""
+    # redact question
+    mock_question_redactor.return_value = "Tell me about Kubernetes"
```

Comment:
the test doesn't seem to rely on this value so why do we need to have a mock return it?

not clear why we care about setting up a question filter mock at all for this test.

Reply:
It does not need it . Fixed it.